### PR TITLE
Many small fixes

### PR DESCRIPTION
**Pull request by deguix**
Basically only last 2 commits are relevant. Everything else is just merge from upstream (sigh git).

### CODE REVIEW

File: acs/DND.bcs [hunk 1]
```diff
@@ -1146,6 +1143,10 @@ Script "DnD Remove Check Special" (int sp, int extra) CLIENTSIDE {
 		if(ConsolePlayerNumber() != (sp & 0xFFFF) - 1)
 			Thing_Remove(0);
 	}
+	else if(extra == DND_SPECIAL_TALENTCAPSULE) {
+		if(ConsolePlayerNumber() != (sp & 0xFFFF) - 1)
```
Comment (Doruk-Aksoy) [inline acs/DND.bcs:1147]: You can just combine this with the above if statement using ||

File: acs/DnD.h [hunk 2]
```diff
@@ -549,7 +554,7 @@ void DistributeBonus(int bonustype) {
 	int bval = 0, temp = 0, i = 0;
 	if(bonustype == BONUS_KILL) {
 		bval = CalculateBonus(BONUS_KILL);
-		temp = GetStat(STAT_LVLEXP) * bval / 100;
+		temp = LevelCurve[GetStat(STAT_LVL)] * bval / 100;
```
Comment (Doruk-Aksoy) [inline acs/DnD.h:557]: This was intentionally set up this way, players get a % of the exp they acquired in that map (wording is confusing here I guess) not their own level.

**Review by Doruk-Aksoy — CHANGES_REQUESTED** [review]

File: acs/DnD.h (hunk 2 above)
Comment (deguix) [reply acs/DnD.h:557]: Without it, players won't get any exp. Nothing in the code uses STAT_LVLEXP.

File: acs/DND.bcs (hunk 1 above)
Comment (deguix) [reply acs/DND.bcs:1147]: you're right.

**Pull request merged**